File: src/dotnet/NinjaSurvivalRate/src/Ninja.cs
using System;
using System.Collections.Generic;

namespace NinjaSurvivalRate
{
    public class Ninja
    {
        public Ninja(String beltLevel)
        {
            Belt = beltLevel;
        }

        public List<String> AttackedBy(String opponent)
        {
            if ("Chuck Norris" == opponent)
                return new List<string>(
                    new[] {"run for his life"});

            return new List<string>(
                new[] {"engage the opponent"});
        }

        public string Belt { get; set; }

        public string CalculateImpact(string opponent)
        {
            if (Belt != "third")
                return "split";
            return opponent == "Chuck Norris" ? "split" : "not harmed";
        }
 
    }
}

File: src/java/NinjaSurvivalRateWithSharedState/src/main/java/ninjasurvivalrate/Ninja.java
package ninjasurvivalrate;
import java.util.Arrays;
import java.util.Collection;
public class Ninja {
	private String belt;
	public Ninja (String belt){
		this.belt=belt;
	}
	public Collection<String> attackedBy(String opponent){
		if ("Chuck Norris".equals(opponent))
			return Arrays.asList(new String[]{"run for his life"});
		else 
			return Arrays.asList(new String[]{"engage the opponent"});
	}
	public String impact(String target,String opponent,String weapon){
		if (!belt.equals("third")) return "split";
		if (opponent.equals("Chuck Norris")) return "split";
		return "not harmed";
	}
}
